src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -1585,6 +1597,104 @@ public ReportType updateCounts( final CodonTracker codonTracker,
+            final int nElements = elements.size();
+            if ( nElements < 2 ) return cigar;
+            final int initialClipLength = getClipLength(elements.get(0));
+            final int finalClipLength = getClipLength(elements.get(nElements - 1));
+            if ( initialClipLength >= minAltLength || finalClipLength >= minAltLength ) {
+                final String saTag = read.getAttributeAsString("SA");
+                if ( saTag == null ) return cigar;
+                final int readLength = read.getLength();
+                final Interval primaryReadInterval = new Interval(initialClipLength, readLength - finalClipLength);
+                for ( final String alt : saTag.split(";") ) {
+                    final String[] fields = alt.split(",");
+                    if ( fields.length != 6 ) {
+                        staticLogger.warn("Badly formed supplemental alignment: " + alt);
+                        continue;
+                    }
+                    try {
+                        if ( Integer.parseInt(fields[4]) < minMapQ ) continue;
+                    } catch ( final NumberFormatException nfe ) {
+                        staticLogger.warn("Can't get mapQ from supplemental alignment: " + alt);
+                        continue;
+                    }
+                    if ( !(read.isReverseStrand() ? "-" : "+").equals(fields[2]) ) continue;
+                    final Cigar altCigar;
+                    try {
+                        altCigar = TextCigarCodec.decode(fields[3]);
+                    } catch ( final IllegalArgumentException iae ) {
+                        staticLogger.warn("Can't parse cigar in supplemental alignment: " + alt);
+                        continue;
+                    }
+                    final List<CigarElement> altElements = altCigar.getCigarElements();
+                    final int nAltElements = altElements.size();
+                    if ( nAltElements < 2 ) continue;
+                    final int initialAltClipLength = getClipLength(altElements.get(0));
+                    final int finalAltClipLength = getClipLength(altElements.get(nAltElements - 1));
+                    final Interval altReadInterval =
+                            new Interval(initialAltClipLength, readLength - finalAltClipLength);
+                    final int overlapLength = primaryReadInterval.overlapLength(altReadInterval);
+                    if ( Math.abs(overlapLength) <= 2 ) {
+                        final int altRefStart;
+                        try {
+                            altRefStart = Integer.parseInt(fields[1]) - 1;
+                        } catch ( final NumberFormatException nfe ) {
+                            staticLogger.warn("Can't parse starting coordinate from supplement alignment: " + alt);
+                            continue;
+                        }
+                        if ( initialClipLength < initialAltClipLength ) {
```

Comment:
is this assuming that alt always comes after primary?

Reply:
no, it's just establishing whether the primary or the supplementary alignment is earlier on the read.  there's an else that takes care of the other case.